modules/openapi-generator/src/test/resources/3_0/oneOf.yaml

```diff
@@ -22,6 +22,9 @@ components:
          oneOf:
             - $ref: '#/components/schemas/apple'
             - $ref: '#/components/schemas/banana'
+         # additionalProperties:
+         #    type: string
+         # uncomment this when https://github.com/swagger-api/swagger-parser/issues/1252 is resolved
```

Comment:
I wouldn't imagine this will get fixed anytime soon in swagger-parser for the 3.0 specification (the revision versions 3.0.1/3.0.2 are supposedly documentation-only changes).

The behavior you're seeing where this returns `MapSchema` is defined behavior: https://github.com/OAI/OpenAPI-Specification/blob/master/versions/3.0.2.md#model-with-mapdictionary-properties

The requested behavior is undefined, as there's not enough clarity in the specification around constraints of using `oneOf`. It _should_ be that `oneOf` can only coexist with `discriminator`.

`oneOf` is taken from JSON Schema's specification [Section 9.2.1.3](https://json-schema.org/draft/2019-09/json-schema-core.html#rfc.section.9.2.1.3), which says:

> This keyword's value MUST be a non-empty array. Each item of the array MUST be a valid JSON Schema.
> 
> An instance validates successfully against this keyword if it validates successfully against exactly one schema defined by this keyword's value.

Meaning your example is only a valid object if it is exactly `apple` or exactly `banana`, and no dynamic properties via `additionalProperties` are allowed due to the use of `MUST` rather than `SHOULD` in the JSON Schema specification.

The `discriminator` support for `oneOf` is a JSON Schema extension defined in the OAS [Composition and Inheritance (Polymorphism)](https://github.com/OAI/OpenAPI-Specification/blob/master/versions/3.0.2.md#composition-and-inheritance-polymorphism) section.

Ideally, a spec with `oneOf` any _anything_ besides discriminator should fail validation. We could consider adding this strict validation to OpenAPI Generator, or at least as a recommendation via the `--recommend` flag to the validate command.